### PR TITLE
Close ipc connection when toggling forno

### PR DESCRIPTION
**Pull request by yorhodes**
### Description

Reinitialize all state which relies on the IPC connection whenever data saver mode is toggled (and geth is restarted with a different configuration)

### Other changes

- Remove `web3forUtils`
- Remove multiple stateful instances of `ContractKit`

### Tested

Tested on `alfajoresstaging` booting with data saver mode off and toggling to on and vice versa

### Related issues

- Fixes bootloop

### Backwards compatibility

_Brief explanation of why these changes are/are not backwards compatible._

### CODE REVIEW

**Review by jmrossy — COMMENTED** [review]
@yorhodes Please fill in the PR template and have a look at the failing mobile test

File: packages/mobile/src/geth/geth.ts [hunk 1]
```diff
@@ -131,18 +126,26 @@ async function initGeth(sync: boolean = true) {
     }
     const geth = await createNewGeth(sync)
 
+    if (!sync) {
+      // TODO: cache somewhere
```
Comment (annakaz) [inline packages/mobile/src/geth/geth.ts]: Can you add more context to this TODO

File: packages/mobile/src/geth/geth.ts [hunk 2]
```diff
@@ -131,18 +126,26 @@ async function initGeth(sync: boolean = true) {
...
+      await deleteChainData()
+      // await deleteNodeDbIfExists()
```
Comment (annakaz) [inline packages/mobile/src/geth/geth.ts]: Remove if no longer used?

File: packages/mobile/src/geth/geth.ts [hunk 3]
```diff
@@ -254,16 +255,24 @@ async function writeStaticNodes(nodeDir: string, enodes: string) {
   console.info(`writeStaticNodes enodes are "${enodes}"`)
   const staticNodesFile = getStaticNodesFile(nodeDir)
   await RNFS.mkdir(getFolder(staticNodesFile))
+  await deleteFileIfExists(staticNodesFile)
   await RNFS.writeFile(staticNodesFile, enodes, 'utf8')
 }
 
-async function attemptGethCorruptionFix(geth: any) {
+// async function deleteNodeDbIfExists() {
```
Comment (annakaz) [inline packages/mobile/src/geth/geth.ts]: Can you add a comment explaining why this is commented out if you want to leave it in

File: packages/mobile/src/web3/saga.ts [hunk 4]
```diff
@@ -129,8 +134,9 @@ export function* waitForWeb3Sync() {
 }
 
 export function* waitWeb3LastBlock() {
+  yield call(waitForGethConnectivity)
```
Comment (annakaz) [inline packages/mobile/src/web3/saga.ts:137]: If in forno mode will this hang indefinitely?

Comment (yorhodes) [reply packages/mobile/src/web3/saga.ts:137]: no, the connection is set once geth starts up
ideally we would monitor to ensure the IPC remains open

**Review by cmcewen — APPROVED** [review]

**Pull request merged**